lightning/src/ln/channelmanager.rs

```diff
@@ -6035,25 +6214,46 @@ where
 			});
 
 			if let Some(height) = height_opt {
+				macro_rules! time_out_htlc {
+					($amt_msat: expr, $payment_hash: expr, $prev_hop_data: expr) => {
+						let mut htlc_msat_height_data = byte_utils::be64_to_array($amt_msat).to_vec();
+						htlc_msat_height_data.extend_from_slice(&byte_utils::be32_to_array(height));
+
+						timed_out_htlcs.push((HTLCSource::PreviousHopData($prev_hop_data), $payment_hash, HTLCFailReason::Reason {
+							failure_code: 0x4000 | 15,
+							data: htlc_msat_height_data
+						}, HTLCDestination::FailedPayment { payment_hash: $payment_hash }));
+					}
+				}
 				channel_state.claimable_htlcs.retain(|payment_hash, (_, htlcs)| {
 					htlcs.retain(|htlc| {
 						// If height is approaching the number of blocks we think it takes us to get
 						// our commitment transaction confirmed before the HTLC expires, plus the
 						// number of blocks we generally consider it to take to do a commitment update,
 						// just give up on it and fail the HTLC.
 						if height >= htlc.cltv_expiry - HTLC_FAIL_BACK_BUFFER {
-							let mut htlc_msat_height_data = byte_utils::be64_to_array(htlc.value).to_vec();
-							htlc_msat_height_data.extend_from_slice(&byte_utils::be32_to_array(height));
-
-							timed_out_htlcs.push((HTLCSource::PreviousHopData(htlc.prev_hop.clone()), payment_hash.clone(), HTLCFailReason::Reason {
-								failure_code: 0x4000 | 15,
-								data: htlc_msat_height_data
-							}, HTLCDestination::FailedPayment { payment_hash: payment_hash.clone() }));
+							time_out_htlc!(htlc.value, *payment_hash, htlc.prev_hop.clone());
 							false
 						} else { true }
 					});
 					!htlcs.is_empty() // Only retain this entry if htlcs has at least one entry.
 				});
+
+				let mut intercepted_htlcs = self.pending_intercepted_htlcs.lock().unwrap();
+				intercepted_htlcs.retain(|_, htlc| {
+					if height >= htlc.forward_info.outgoing_cltv_value - HTLC_FAIL_BACK_BUFFER {
```

Comment:
They would only be able to drown our node in interception requests if the user isn't handling `HTLCIntercepted` events as they come in, IIUC, but I feel like since the feature is explicitly opt-in dunno if we should be concerned about that(?)

If a user takes several hours to open a channel, I think from the docs they should be aware that there's a risk the HTLC will be cancelled back. I'd be pretty hesitant to add a boiler-platey event.

Reply:
For now the feature is opt-in, and the first concern should be addressed with a `MAX_PENDING_INTERCEPTED_HTLC`. For the second concern, I think we can update the doc of `forward_intercepted_htlc` to underscore this risk of asynchronicity between the intercepting module and the LDK node. I know there is already "Errors if the event was not handled in time, in which case the HTLC was automatically failed", up to you.